cli/src/cmd/inspect.rs

```diff
@@ -0,0 +1,251 @@
+use std::{fmt, str::FromStr};
+
+use crate::{
+    cmd::{
+        build::{self, BuildArgs},
+        Cmd,
+    },
+    opts::forge::CompilerArgs,
+};
+use clap::Parser;
+use ethers::prelude::artifacts::output_selection::{ContractOutputSelection, EvmOutputSelection};
+use serde_json::{to_value, Value};
+
+/// Contract level output selection
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum ContractArtifactFields {
+    Abi,
+    Bytecode,
+    DeployedBytecode,
+    Assembly,
+    AssemblyOptimized,
+    MethodIdentifiers,
+    GasEstimates,
+    StorageLayout,
+    DevDoc,
+    Ir,
+    IrOptimized,
+    // TODO: Add once these commands aren't broken at compiler-level
+    // Metadata,
+    // UserDoc,
+    // Ewasm,
+}
+
+impl fmt::Display for ContractArtifactFields {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ContractArtifactFields::Abi => f.write_str("abi"),
+            ContractArtifactFields::Bytecode => f.write_str("bytecode"),
+            ContractArtifactFields::DeployedBytecode => f.write_str("deployedBytecode"),
+            ContractArtifactFields::Assembly => f.write_str("assembly"),
+            ContractArtifactFields::AssemblyOptimized => f.write_str("assemblyOptimized"),
+            ContractArtifactFields::MethodIdentifiers => f.write_str("methodIdentifiers"),
+            ContractArtifactFields::GasEstimates => f.write_str("gasEstimates"),
+            ContractArtifactFields::StorageLayout => f.write_str("storageLayout"),
+            ContractArtifactFields::DevDoc => f.write_str("devdoc"),
+            ContractArtifactFields::Ir => f.write_str("ir"),
+            ContractArtifactFields::IrOptimized => f.write_str("irOptimized"),
+            // TODO: Add once these commands aren't broken at compiler-level
+            // ContractArtifactFields::Metadata => f.write_str("metadata"),
+            // ContractArtifactFields::UserDoc => f.write_str("userdoc"),
+            // ContractArtifactFields::Ewasm => f.write_str("ewasm"),
```

Comment:
Perhaps the type bindings are wrong on ethers?
can you provide a failing example so we can check what we need to change

Reply:
added back